cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -55,6 +55,7 @@ using internal::BinaryBitBlockCounter;
 using internal::BitBlockCount;
 using internal::BitmapReader;
 using internal::checked_cast;
+using internal::checked_pointer_cast;
```

Comment:
why in header?

Reply:
I was following the pattern of the `internal::checked_cast` on the line above. But I'm not wedded to including it.